File: app/src/main/java/com/leodeleon/popmovies/di/component/ViewModelSubComponent.java
package com.leodeleon.popmovies.di.component;

import com.leodeleon.popmovies.feature.viewModel.MovieDetailsViewModel;
import com.leodeleon.popmovies.feature.viewModel.MoviesViewModel;
import dagger.Subcomponent;

@Subcomponent public interface ViewModelSubComponent {
  MoviesViewModel moviesViewModel();
  MovieDetailsViewModel movieDetailsViewModel();

  @Subcomponent.Builder interface Builder {
    ViewModelSubComponent build();
  }
}

File: app/src/main/java/com/leodeleon/popmovies/di/component/ViewModelSubComponent.kt
package com.leodeleon.popmovies.di.component

import com.leodeleon.popmovies.feature.viewModel.MovieDetailsViewModel
import com.leodeleon.popmovies.feature.viewModel.MoviesViewModel
import dagger.Subcomponent

@Subcomponent interface ViewModelSubComponent {

  fun moviesViewModel(): MoviesViewModel

  fun movieDetailsViewModel(): MovieDetailsViewModel

  @Subcomponent.Builder interface Builder {
    fun build(): ViewModelSubComponent
  }
}
